boilerplate/app/components/ListItem.tsx

```diff
@@ -144,9 +144,13 @@ export function ListItem(props: ListItemProps) {
           Component={LeftComponent}
         />
 
-        <Text {...TextProps} tx={tx} text={text} txOptions={txOptions} style={$textStyles}>
-          {children}
-        </Text>
+        {!!text || !!tx || typeof children === "string" ? (
+          <Text {...TextProps} tx={tx} text={text} txOptions={txOptions} style={$textStyles}>
+            {children}
+          </Text>
+        ) : (
+          children
```

Comment:
Looks good, do we need any View styling around this or should we just let the user take care of that with what they're passing in?

$textStyles has some styling associated with it that we'd be missing here.

If we do add that wrapper though, then we'd need some sort of container override, so maybe better not to.

Thoughts?

Reply:
I think it makes more sense to just let the user style the children directly. The original issue the client had with this component was that `$textStyles` unintuitively styled the children being passed in so they were trying all sorts of different styles and getting unexpected results. I have a low gafo on this but I think adding additional styling would just result in the same issue.

Reply:
@kateinkim Sounds good to me!

Reply:
There's a potential deprecation here with not allowing to pass other Text children into the <Text /> (Text's are meant to be nestable here). I'm wondering if there's a better way to check what type `children` is without adding another prop... Thinking...

Reply:
I can just leave this PR for adding const assertion to the color palette and open an issue for the ListItem?